packages/event-props/src/index.ts

```diff
@@ -0,0 +1,43 @@
+import { createSignal } from "solid-js";
+
+export type HTMLEventName = keyof HTMLElementEventMap
+export type EventNames = [name: HTMLEventName, ...moreNames: HTMLEventName[]];
+export type EventStore<Names extends HTMLEventName[]> = {
+  [name in Names[number]]?: HTMLElementEventMap[name]
+};
+export type EventProps<Names extends EventNames> = {
+  [name in Names[number] as `on${name}`]: (ev: HTMLElementEventMap[name]) => void
+};
+
+/**
+ * Creates the props for certain events for an intrinsic JSX element and a store with the received events
+ * @param names of the events that should be available in props and store
+ * @returns [eventStore, eventProps]
+ * @example
+ * ```ts
+ * const [events, eventProps] = createEventProps('mousedown', 'mousemove', 'mouseup');
+ * 
+ * const isMouseDown = createMemo(() => (events.mousedown?.ts ?? 0) > (events.mouseup?.ts ?? 1));
+ * 
+ * createEffect(() => {
+ *   if (isMouseDown) {
+ *     console.log(events.mousemove?.clientX, events.mousemove?.clientY);
+ *   }
+ * });
+ * 
+ * <div {...eventProps}>Click and drag on me</div>
+ * ```
+ */
+export const createEventProps = <Names extends EventNames>(
+  ...names: Names
+): [EventStore<Names>, EventProps<Names>] => {
+  type EventStore = Partial<Pick<HTMLElementEventMap, typeof names[number]>>;
```

Comment:
I find it strange to declare types within, shouldn't this be outside the function and exported?

Reply:
Ah, that somehow snuck in. Sorry, fixed it.